pkg/testing/fixture.go

```diff
@@ -724,16 +724,23 @@ func (f *Fixture) ExecVersion(ctx context.Context, opts ...process.CmdOption) (A
 	return version, err
 }
 
-// IsHealthy returns if the prepared Elastic Agent reports itself as healthy.
-// It returns false, err if it cannot determine the state of the agent.
-// It should work with any 8.6+ agent
+// IsHealthy checks whether the prepared Elastic Agent reports itself as healthy.
+// It returns false and an error if either the reported state isn't healthy or
+// if it fails to fetch the current state. If the status is successfully fetched
+// but isn't healthy, the error will contain the reported status.
+// This function is compatible with any Elastic Agent version 8.6 or later.
 func (f *Fixture) IsHealthy(ctx context.Context, opts ...process.CmdOption) (bool, error) {
 	status, err := f.ExecStatus(ctx, opts...)
 	if err != nil {
 		return false, fmt.Errorf("agent status returned and error: %w", err)
 	}
 
-	return status.State == int(cproto.State_HEALTHY), nil
+	if status.State != int(cproto.State_HEALTHY) {
+		return false, fmt.Errorf("agent isn't health, current status: %s",
+			client.State(status.State))
```

Comment:
I understand the reason for adding this, but this kinda changes the meaning of the error in this function. The error was more that it failed to get the actual health. Now this changes it to return an error when its not healthy, that is probably not was the caller is expecting.

Reply:
This is true, but it's also why I changed it docs. Besides there is just one usage of it, that is the test which needed this change. I could change the function signature, but I believe the doc change clarifies what is returned in the error.

Reply:
Based on the way your changing it, would you be willing to change it to `EnsureHealthy` and remove the boolean? Doesn't seem like true/false is relevant to what this function is even doing at more, so just have it return error or nil.

Reply:
It does make sense, I just kept the name because it isn't ensuring anything, just checking the status